sim/shaman/elemental/dragonwrath.go

```diff
@@ -8,12 +8,19 @@ import (
 )
 
 func init() {
+	//TODO (Chain Lightning) I think multiple overloads dup are possible, but couldn't
+	//observe any during testing because of low chance of it happening ?
+	//DTR proc also do not always start on the same target the base spell did
+	//so DPS on a specific target might not be accurate.
 	cata.CreateDTRClassConfig(proto.Spec_SpecElementalShaman, 0.108).
-		AddSpell(88767, cata.NewDragonwrathSpellConfig().WithSpellHandler(customFulminationHandler)). // Fullmination
-		AddSpell(403, cata.NewDragonwrathSpellConfig().WithCustomSpell(overloadCopyHandler)).         // Lightning Bold
-		AddSpell(421, cata.NewDragonwrathSpellConfig().WithCustomSpell(overloadCopyHandler)).         // Chain Lightning
-		AddSpell(51505, cata.NewDragonwrathSpellConfig().WithCustomSpell(overloadCopyHandler)).       // Lava Burst
-		AddSpell(3599, cata.NewDragonwrathSpellConfig().SupressSpell())                               // Searing Totem
+		AddSpell(88767, cata.NewDragonwrathSpellConfig().WithSpellHandler(customFulminationHandler)).                              // Fullmination
+		AddSpell(403, cata.NewDragonwrathSpellConfig().WithCustomSpell(overloadCopyHandler)).                                      // Lightning Bold
+		AddSpell(421, cata.NewDragonwrathSpellConfig().WithCustomSpell(overloadCopyHandler).ProcPerCast()).                        // Chain Lightning
+		AddSpell(51505, cata.NewDragonwrathSpellConfig().WithCustomSpell(overloadCopyHandler)).                                    // Lava Burst
+		AddSpell(3599, cata.NewDragonwrathSpellConfig().SupressSpell()).                                                           // Searing Totem
+		AddSpell(8190, cata.NewDragonwrathSpellConfig().SupressSpell()).                                                           // Magma Totem
+		AddSpell(51490, cata.NewDragonwrathSpellConfig().ProcPerCast()).                                                           // Thunderstorm
+		AddSpell(77478, cata.NewDragonwrathSpellConfig().IsAoESpell().TreatTickAsCast().WithSpellHandler(customEarthquakeHandler)) // Earthquake
```

Comment:
Looking at how EQ is implemented I'd suggest that you don't even need any fancy custom handling from DTR.
You could simply remodel the spell like I did for Druid's mushrooms here:

https://github.com/wowsims/cata/commit/fea603605283b8e1bcf9e9fdd3739ad7255a7841#diff-193a3c2777a162ceb4599f49352b2c5cbd6897dbe5589811348df4490871590c

As from my obersevation each tick is simply a an AoE cast like mind sear or shrooms that gets replicated as a whole. You'll still need to mark it as TreatTickasCast when you want to deal PeriodicDamage with it though

https://classic.warcraftlogs.com/reports/aHYJCbRBzK3rFfqT#boss=-3&difficulty=0&type=damage-done&start=7555099&end=7558099&ability=77478&view=events

Should show that the whole tick is replicated.

Reply:
i think he got lucky and had 3 replicated tick at once. here's my log from testing : https://classic.warcraftlogs.com/reports/XckVzbHgYyD6JtKP#boss=-3&difficulty=0&source=22&type=damage-done&ability=77478&start=1848649&end=2529940&view=events on 3 targets

Reply:
> Should show that the whole tick is replicated.

In fact, its not even the same targets hit by the replicated spells, i'm gonna ask him where he was testing